proxy/src/main/java/com/velocitypowered/proxy/server/VelocityRegisteredServer.java

```diff
@@ -83,20 +84,27 @@ public Collection<Player> getPlayersConnected() {
     return ImmutableList.copyOf(players.values());
   }
 
+  @Override
+  public CompletableFuture<ServerPing> ping(PingOptions pingOptions) {
+    return ping(null, pingOptions.getTimeout(), pingOptions.getProtocolVersion());
+  }
+
   @Override
   public CompletableFuture<ServerPing> ping() {
-    return ping(null, ProtocolVersion.UNKNOWN);
+    return ping(null, -1, ProtocolVersion.UNKNOWN);
   }
 
   /**
    * Pings the specified server using the specified event {@code loop}, claiming to be
    * {@code version}.
    *
    * @param loop    the event loop to use
-   * @param version the version to report
+   * @param timeout the timeout respective
+   * @param protocolVersion the version to report
    * @return the server list ping response
    */
-  public CompletableFuture<ServerPing> ping(@Nullable EventLoop loop, ProtocolVersion version) {
+  public CompletableFuture<ServerPing> ping(
+          @Nullable EventLoop loop, long timeout, ProtocolVersion protocolVersion) {
```

Comment:
To clean this a bit, we can make the no-arguments `ping` method pass a default `PingOptions` instance (which can be cached and be exposed as an static member called `DEFAULT` in that class) to this method, and remove the method with all options below (which won't "scale" as the option set grows in the future).